tests/unit/test_dunamai.py

```diff
@@ -498,16 +498,12 @@ def test__check_version__semver() -> None:
         Version("00.0.0").serialize(style=style)
     with pytest.raises(ValueError):
         Version("0.01.0").serialize(style=style)
-    with pytest.raises(ValueError):
-        Version("0.1.0-alpha.02").serialize(style=style)
```

Comment:
These are important tests and should not be removed. If these failed with the new code, then the new code has an issue.

Reply:
I removed them cause by processing the base string this version serialises  to a valid string. Optionally you could add test without the `ValueError`

Reply:
Ah, I see; since the revision is getting parsed as an int, it strips the leading zero. I thought it was actually serializing as `02`, which would've been invalid for Semantic Versioning.